### PR TITLE
[MM-28216] Hookups to back-end for Billing History section

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
This contains all the hookups to Redux to pull down billing history. It also contains a virtual paging implementation that can be converted once the server supports actual paging.

UX has already been reviewed by Matt.
Will need QA once hooked up to CWS.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28216

### CODE REVIEW

File: components/admin_console/billing/billing_history.tsx [hunk 1]
```diff
@@ -94,62 +63,90 @@ const getPaymentStatus = (status: string) => {
...
+    };
 
     useEffect(() => {
+        dispatch(getCloudProducts());
+        dispatch(getCloudSubscription());
+        dispatch(getInvoices());
         pageVisited('cloud_admin', 'pageview_billing_history');
     }, []);
 
+    useEffect(() => {
```
Comment (ethervoid) [inline components/admin_console/billing/billing_history.tsx:123]: Why use `useEffect` two times?

**Review by ethervoid — COMMENTED** [review]
One question

File: components/admin_console/billing/billing_history.tsx [hunk 2]
```diff
@@ -186,43 +183,59 @@ const BillingHistory: React.FC<Props> = () => {
...
+                                        values={{
+                                            fullUsers,
+                                            partialUsers,
+                                        }}
+                                    />
+                                </div>
+                            </td>
+                            <td className='BillingHistory__table-total'>
+                                <FormattedNumber
+                                    value={invoice.total}
```
Comment (ethervoid) [inline components/admin_console/billing/billing_history.tsx:218]: Invoice total comes as an int where the last two digits are the decimal so we need

File: components/admin_console/billing/billing_history.tsx (hunk 1 above)
Comment (devinbinnie) [reply components/admin_console/billing/billing_history.tsx:123]: First `useEffect` only runs once as there are no dependent parameters (see the second argument), so it acts like `componentDidMount`.

Second `useEffect` updates other state when the two dependent parameters change.

Comment (ethervoid) [reply components/admin_console/billing/billing_history.tsx:123]: Cool, good to know :)

**Review by ethervoid — APPROVED** [review]
LGTM!

File: components/admin_console/billing/billing_history.tsx [hunk 3]
```diff
@@ -94,62 +63,90 @@ const getPaymentStatus = (status: string) => {
...
+        dispatch(getCloudProducts());
+        dispatch(getCloudSubscription());
+        dispatch(getInvoices());
         pageVisited('cloud_admin', 'pageview_billing_history');
     }, []);
 
+    useEffect(() => {
+        if (invoices && Object.values(invoices).length) {
+            const invoicesByDate = Object.values(invoices).sort((a, b) => b.period_start - a.period_start);
+            setBillingHistory(invoicesByDate.slice(firstRecord - 1, (firstRecord - 1) + PAGE_LENGTH));
```
Comment (marianunez) [inline components/admin_console/billing/billing_history.tsx:126]: Wondering if making `firstRecord` zero-based would simplify some of these calculations and below when setting the paging indicators.

**Review by marianunez — APPROVED** [review]
Great work, thanks @devinbinnie!

File: components/admin_console/billing/billing_history.tsx (hunk 3 above)
Comment (devinbinnie) [reply components/admin_console/billing/billing_history.tsx:126]: We could, but then we'd need to do the same in reverse for displaying the record counts. Might reduce some code bloat but it'd be negligible. 0/5

Comment (marianunez) [reply components/admin_console/billing/billing_history.tsx:126]: Agreed

**Pull request merged**